v2/cmd/zlint/main.go

```diff
@@ -210,3 +220,83 @@ func setLints() (lint.Registry, error) {
 
 	return lint.GlobalRegistry().Filter(filterOpts)
 }
+
+func outputSummary(zlintResult *zlint.ResultSet, longSummary bool) {
```

Comment:
I think some of this summary code could be hoisted into the `zlint.ResultSet` type. I think there's some similarity here with https://github.com/zmap/zlint/issues/384 for example. The textual summary parts should stay here of course :-)

Reply:
It would also probably help write unit tests for the code. I think this logic is intricate enough to merit a few unit tests and it will probably be easier to test outside of the CLI app.

Reply:
thanks @cpu 

I'm less comfortable messing in `zlint.ResultSet` but would give it a try if you want.

I can also add some unit tests and update this PR, it will be a little while before I have time, though.

Reply:
> I'm less comfortable messing in zlint.ResultSet but would give it a try if you want.

As an alternative, what about breaking out a `struct` type that lives in this package's scope for now? I think `outputSummary` is quite long as-is and would benefit from a little bit of refactoring. It's doing three things right now: initializing some related state, displaying short output, and displaying long output. Breaking up the initialization and the display seems like a good place to start.

Off the bat I think there's a bit of related state that could be grouped into struct members: `resultCount`, `resultDetails`, `lintLevelsAboveThreshold` and `sortedLevels`. Something like:

```go
type resultsTable struct {
  resultCount map[lint.LintStatus]int
  resultDetails map[lint.lintStatus][]string
  lintLevelsAboveThreshold map[int]lint.Status
  sortedLevels []int
}
```

You could then split out all of the `map` initialization from `outputSummary` into a constructor for the `resultsTable` type, something like:
```go
func newResultsTable(threshold lint.LintResult, results *zlint.ResultSet) resultsTable {
  // init struct members here w/ loops from the `outputSummary` code
  ...
}
```

That would remove the initialization concern from `outputSummary` and let it be re-implemented as a method on the `resultsTable` type that can strictly concern itself with outputting the results:

```go
func (t resultsTable) outputSummary(longSummary bool) {
   ....
}
```

WDYT?